### DISCUSSION TITLE
Count the percentage of white pixels on image

### DISCUSSION THREAD

**Discussion by gmarcucci1987** [dlemstra/Magick.NET#729] [Help] [answered]
Hi,
I was researching a way to calculate the percentage of white pixels on an image. And found this example on image magick, I was wondering how would I translate this commands to c#:

convert 50889165.png -fuzz 4% -fill White -opaque White -fuzz 0 -fill Black +opaque White -precision 15 -format %[fx:int(mean*w*h+0.5)] info:

Source:
https://www.imagemagick.org/discourse-server/viewtopic.php?t=28264

**Comment by dlemstra** [comment, marked as answer]
Do you want to get the total of white or white-ish pixels? For the first there is an easier way to do this.

**Reply by gmarcucci1987** [reply]
Percentage of white is good for my use case

**Reply by dlemstra** [reply]
This could work for you:

```C#
var histogram = image.Histogram();
var total = (double)histogram.Values.Sum();
if (histogram.TryGetValue(MagickColors.White, out var totalWhite))
{
    var percentage = totalWhite / total;
}
```

If this doesn't work because you have white pixels that are not full white I can help you translate the command. In case you want to give it a try yourself here is a tip: Most options are methods or properties on the `MagickImage` class.

**Reply by gmarcucci1987** [reply]
Thank it works great.  Sorry for the delay but I'm using that function to check the amount of color in any given image.

Aditionally I'm trying to calculate the intensity or luminasance of an image to see if it's too dark and this command is working for me:

"convert '{fileName}' -colorspace HSI -channel b -separate +channel -scale 1x1 -format '%[fx: 100 * u]\n' info:"

I can translate everything except the last part 

  using (var image = new MagickImage(fileName))
            {
                image.ColorSpace = ColorSpace.HSI;
                using (var intensity = image.Separate(Channels.Blue).First())
                {
                    intensity.Scale(1, 1);
                    intensity.Fx("u * 100");
                }
            }

How can I translate it to get the percentage

**Reply by gmarcucci1987** [reply]
I ended up with this code, which  gave me the same value:

using (var image = new MagickImage(fileName))
            {
                image.Scale(1, 1);
                using (var pixels = image.GetPixels())
                {
                    var color = pixels.GetPixel(0, 0).ToColor();
                    var Hslcolor = ColorHSL.FromMagickColor(color);
                    var lightness = Hslcolor.Lightness*100;
                }

            }

is this approach correct?

**Reply by dlemstra** [reply]
You could also use `image.FormatExpression("%[fx: 100 * u")`.
